### PR TITLE
extend exclusion rules in monitoring report

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/86941dawr

### :memo: Implementation
- Added exclusion rules for `excludedRoles`, `excludedUsers`, `excludedRolesByRole` and `excludedRolesByUser`

### :fire: Notes to the tester
Test against this matrix:
https://docs.google.com/spreadsheets/d/11dnC4BQ-oNA_67bjkCJbws1YRzcfJWk4pnYJ63c4l_k/edit#gid=0

```
REACT_APP_DHIS2_BASE_URL=https://dev.eyeseetea.com/who-dev-238/
REACT_APP_REPORT_VARIANT=authorities-monitoring
```

#86941dawr

### CODE REVIEW

**Comment by idelcano** [comment]
I have merged development and fixed the workflow of the rules, also i have included excludedRolesByGroups
I have created this matrix to test it:
https://docs.google.com/spreadsheets/d/11dnC4BQ-oNA_67bjkCJbws1YRzcfJWk4pnYJ63c4l_k/edit?gid=0#gid=0

and the variant to make it work is:
REACT_APP_REPORT_VARIANT=authorities-monitoring

Its already installed in dev: 
https://extranet.who.int/dhis2-dev/dhis-web-reports/index.html#/standard-report/view/ctfks02sZka

File: src/data/reports/authorities-monitoring/AuthoritiesMonitoringDefaultRepository.ts [hunk 1]
```diff
@@ -156,18 +169,74 @@ export class AuthoritiesMonitoringDefaultRepository implements AuthoritiesMonito
     ): Promise<AuthoritiesMonitoringItem[]> {
         const { templateGroups, usernameQuery, userRoles } = options;
 
+        // eslint-disable-next-line
+        debugger;
```
Comment (nshandra) [inline src/data/reports/authorities-monitoring/AuthoritiesMonitoringDefaultRepository.ts:173]: @idelcano This is leftover code, right?

**Review by nshandra — CHANGES_REQUESTED** [review]
Looks ok, aside from the leftover code mentioned.

**Review by nshandra — APPROVED** [review]
Looks good. Approved.

**Pull request merged**